cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/simple_dataset_selector.spec.js

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  INDEX_PATTERN_WITH_TIME,
+  INDEX_WITH_TIME_1,
+  INDEX_PATTERN_WITH_NO_TIME,
+  INDEX_WITHOUT_TIME_1,
+  SECONDARY_ENGINE,
+} from '../../../../../utils/constants';
+import {
+  getRandomizedWorkspaceName,
+  getRandomizedDatasourceName,
+  getDefaultQuery,
+  setDatePickerDatesAndSearchIfRelevant,
+} from '../../../../../utils/apps/query_enhancements/shared';
+import { verifyDiscoverPageState } from '../../../../../utils/apps/query_enhancements/saved';
+import {
+  generateSimpleDatasetSelectorTestConfigurations,
+  validateItemsInSimpleDatasetSelectorDropDown,
+} from '../../../../../utils/apps/query_enhancements/simple_dataset_selector';
+
+const workspaceName = getRandomizedWorkspaceName();
+const datasourceName = getRandomizedDatasourceName();
+const noIndexPatterns = 5;
+
+export const runSimpleDatasetSelectorTests = () => {
+  describe('simple dataset selector selecting an index pattern', () => {
+    beforeEach(() => {
+      // Load test data
+      cy.setupTestData(
+        SECONDARY_ENGINE.url,
+        [
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITH_TIME_1}.mapping.json`,
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITHOUT_TIME_1}.mapping.json`,
+        ],
+        [
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITH_TIME_1}.data.ndjson`,
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITHOUT_TIME_1}.data.ndjson`,
+        ]
+      );
+      // Add data source
+      cy.addDataSource({
+        name: datasourceName,
+        url: SECONDARY_ENGINE.url,
+        authType: 'no_auth',
+      });
+
+      // Create workspace
+      cy.deleteWorkspaceByName(workspaceName);
+      cy.visit('/app/home');
+      cy.osd.createInitialWorkspaceWithDataSource(datasourceName, workspaceName);
+      cy.createWorkspaceIndexPatterns({
```

Comment:
nit:

for 
```
cy.createWorkspaceIndexPatterns({
        workspaceName: workspaceName,
        indexPattern: INDEX_PATTERN_WITH_TIME.replace('*', ''),
        timefieldName: 'timestamp',
        dataSource: datasourceName,
        isEnhancement: true,
      });
      cy.createWorkspaceIndexPatterns({
        workspaceName: workspaceName,
        indexPattern: INDEX_PATTERN_WITH_NO_TIME.replace('*', ''),
        timefieldName: '',
        dataSource: datasourceName,
        isEnhancement: true,
        indexPatternHasTimefield: false,
      });
```
I think we could move into the permutation generateSimpleDatasetSelectorTestConfigurations. We don't need to create index pattern if test for indexes.

Reply:
nvm I saw generateSimpleDatasetSelectorTestConfigurations is for index pattern only